recipe/userroles/recipeimplementation.go

```diff
@@ -0,0 +1,239 @@
+/* Copyright (c) 2022, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package userroles
+
+import (
+	"errors"
+
+	"github.com/supertokens/supertokens-golang/recipe/userroles/userrolesmodels"
+	"github.com/supertokens/supertokens-golang/supertokens"
+)
+
+func makeRecipeImplementation(querier supertokens.Querier, config userrolesmodels.TypeNormalisedInput, appInfo supertokens.NormalisedAppinfo) userrolesmodels.RecipeInterface {
+
+	addRoleToUser := func(userID string, role string, userContext supertokens.UserContext) (userrolesmodels.AddRoleToUserResponse, error) {
+		response, err := querier.SendPutRequest("/recipe/user/role", map[string]interface{}{
+			"userId": userID,
+			"role":   role,
+		})
+		if err != nil {
+			return userrolesmodels.AddRoleToUserResponse{}, err
+		}
+
+		if response["status"] == "OK" {
+			return userrolesmodels.AddRoleToUserResponse{
+				OK: &struct{ DidUserAlreadyHaveRole bool }{
+					DidUserAlreadyHaveRole: response["didUserAlreadyHaveRole"].(bool),
+				},
+			}, nil
+		}
+
+		return userrolesmodels.AddRoleToUserResponse{
+			UnknownRoleError: &userrolesmodels.UnknownRoleError{},
+		}, nil
+	}
+
+	removeUserRole := func(userID string, role string, userContext supertokens.UserContext) (userrolesmodels.RemoveUserRoleResponse, error) {
+		response, err := querier.SendPostRequest("/recipe/user/role/remove", map[string]interface{}{
+			"userId": userID,
+			"role":   role,
+		})
+		if err != nil {
+			return userrolesmodels.RemoveUserRoleResponse{}, err
+		}
+
+		if response["status"] == "OK" {
+			return userrolesmodels.RemoveUserRoleResponse{
+				OK: &struct{ DidUserHaveRole bool }{
+					DidUserHaveRole: response["didUserHaveRole"].(bool),
+				},
+			}, nil
+		}
+
+		return userrolesmodels.RemoveUserRoleResponse{
+			UnknownRoleError: &userrolesmodels.UnknownRoleError{},
+		}, nil
+	}
+
+	getRolesForUser := func(userID string, userContext supertokens.UserContext) (userrolesmodels.GetRolesForUserResponse, error) {
+		response, err := querier.SendGetRequest("/recipe/user/roles", map[string]string{
+			"userId": userID,
+		})
+		if err != nil {
+			return userrolesmodels.GetRolesForUserResponse{}, err
+		}
+
+		if response["status"] == "OK" {
```

Comment:
You do not need to check for status OK - since it's always going to be that. Therefore, you don't need the `return userrolesmodels.GetRolesForUserResponse{}, errors.New("should never come here")` line either.